src/utils.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::{Path, PathBuf};
+use std::io::Error as IoError;
+use std::env::VarError;
+
+#[derive(Debug)]
+pub enum Error {
+    EnvError(VarError),
+    IoError(IoError),
+}
+
+impl From<VarError> for Error {
+    fn from(e: VarError) -> Self {
+        Error::EnvError(e)
+    }
+}
+
+impl From<IoError> for Error {
+    fn from(e: IoError) -> Self {
+        Error::IoError(e)
+    }
+}
+
+fn not_found_error() -> IoError {
+    use std::io::ErrorKind as IoErrorKind;
+    IoError::new(IoErrorKind::NotFound, "No file name component")
+}
+
+#[allow(dead_code)]
```

Comment:
but this function might not be used in other platforms we choose to support in the future. the rust compiler would refuse to compile the code for no good reason. are you sure?